synapse/rest/client/login.py

```diff
@@ -300,10 +309,15 @@ async def _do_appservice_login(
             # The user represented by an appservice's configured sender_localpart
             # is not actually created in Synapse.
             should_check_deactivated=qualified_user_id != appservice.sender,
+            request_info=request_info,
         )
 
     async def _do_other_login(
-        self, login_submission: JsonDict, should_issue_refresh_token: bool = False
+        self,
+        login_submission: JsonDict,
+        should_issue_refresh_token: bool = False,
+        *,
+        request_info: RequestInfo,
```

Comment:
It's mainly because its a new argument *and* its mandatory. Adding it to the middle of the list (which you'd have to do as the previous args are optional) would mean ensuring that each call site has the correct order. Adding it as a mandatory keyword arg ensures that you can't mess it up.

Reply:
Oh yes I missed the fact that some pos args were optional.